app/api/sponsors.py

```diff
@@ -63,29 +71,52 @@ def query(self, view_kwargs):
             query_ = query_.join(Event).filter(Event.id == event.id)
         return query_
 
-    def before_create_object(self, data, view_kwargs):
+    methods = ['POST']
+    schema = SponsorSchema
+    data_layer = {'session': db.session,
+                  'model': Sponsor}
+
+
+class SponsorList(ResourceList):
+    """
+    List Sponsors
+    """
+
+    def query(self, view_kwargs):
         """
-        method to create object before post
-        :param data:
+        query method for Sponsor List
         :param view_kwargs:
         :return:
         """
+        query_ = self.session.query(Sponsor)
         if view_kwargs.get('event_id'):
             event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-            data['event_id'] = event.id
-
+            if event.state != 'published':
+                if 'Authorization' in request.headers and has_access('is_coorganizer', event_id=event.id):
+                    query_ = query_.join(Event).filter(Event.id == event.id)
+                else:
+                    raise ObjectNotFound({'parameter': 'event_id'},
+                                         "Event: {} not found".format(view_kwargs['event_identifier']))
+            else:
+                query_ = query_.join(Event).filter(Event.id == event.id)
         elif view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'identifier')
-            data['event_id'] = event.id
+            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
```

Comment:
Just a suggestion, can we first get event_id if `view_kwargs['event_identifier']` is present, then write the query block only for 
`if view_kwargs.get('event_id'):` ?